Formula/g/gdrive-cli.rb

```diff
@@ -0,0 +1,22 @@
+class GdriveCli < Formula
+  desc "Google Drive CLI Client"
+  homepage "https://github.com/glotlabs/gdrive"
+  url "https://github.com/glotlabs/gdrive/releases/download/3.9.0/gdrive_macos-x64.tar.gz"
+  sha256 "348f57f42785e3eedfb933bcf079deba8cf51ac2cf699d29539f33956d025d57"
+  license "MIT"
+  head "https://github.com/glotlabs/gdrive.git", branch: "main"
+
+  bottle do
+    sha256 cellar: :any_skip_relocation, all: "aedaf65cf84ca164aa46149a14df2cab9a3ff9d7679fea4621442170839f3e0b"
+  end
+
+  conflicts_with "gdrive", because: "gdrive also ships a gdrive binary"
+
+  def install
+    bin.install "gdrive"
+  end
+
+  test do
+    assert_match "gdrive 3.9.0", shell_output("#{bin}/gdrive version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Noted! I shall update with improvements shortly after reading more.